### ISSUE TITLE
grails 6.0.0 and create-app broken on windows

### ISSUE THREAD

**Issue by chrisbitmead** [grails/grails-core#13112]
### Expected Behavior

Firstly, I installed grails via
```sdk install grails 6.0.0```

and the resulting grails command didn't have execute permission:

```
$ /c/Users/chris/.sdkman/candidates/grails/current/bin/grails -v
bash: /c/Users/chris/.sdkman/candidates/grails/current/bin/grails: Permission denied
```

with cygwin bash. Never seen that before with long grails use. Anyway, I did chmod +x, and then

```
grails create-app myApp600
cd myApp600
grails run-app
Unmatched argument at index 0: 'run-app'
Did you mean: grails create-app or grails create-webapp or grails create-restapi?
```

Never seen that before in long grails usage. I'm running cygwin here, but hard to see how cygwin is involved since grails is a binary.

Then tried this:

```
$ ./gradlew bootWar
./gradlew: line 2: $'\r': command not found
./gradlew: line 18: $'\r': command not found
./gradlew: line 64: $'\r': command not found
./gradlew: line 66: $'\r': command not found
./gradlew: line 69: $'\r': command not found
./gradlew: line 71: $'while\r': command not found
./gradlew: line 73: [: missing `]'
./gradlew: line 74: $'do\r': command not found
ls: cannot access './gradlew'$'\r': No such file or directory
./gradlew: line 79: syntax error near unexpected token `newline'
'/gradlew: line 79: `      *)    app_path=$APP_HOME$link ;;
```

Seems like grails create-app managed to create shell scripts with the wrong line endings. Again, I've been using grails since v3 on windows, and never seen that.

### Actual Behaviour

Should behave on windows

### Environment Information

Windows 11

### Example Application

grails create-app myApp

### Version

6.0.0

**Comment by puneetbehl** [comment]
The Grails framework is moved from the command related to build stuff (such as `grails run-app`, `grails test-app`, etc). Instead, we rely on the Gradle for such things. We would recommend you to use `./gradlew bootRun` to run the application.

Also, I am not sure if the following error is related to Grails:


```
$ ./gradlew bootWar
./gradlew: line 2: $'\r': command not found
./gradlew: line 18: $'\r': command not found
./gradlew: line 64: $'\r': command not found
./gradlew: line 66: $'\r': command not found
./gradlew: line 69: $'\r': command not found
./gradlew: line 71: $'while\r': command not found
./gradlew: line 73: [: missing `]'
./gradlew: line 74: $'do\r': command not found
ls: cannot access './gradlew'$'\r': No such file or directory
./gradlew: line 79: syntax error near unexpected token `newline'
'/gradlew: line 79: `      *)    app_path=$APP_HOME$link ;;
```

Can you try `./gradlew.bat bootRun`?

**Comment by chrisbitmead** [comment]
> Also, I am not sure if the following error is related to Grails:
> 
> ```
> $ ./gradlew bootWar
> ./gradlew: line 2: $'\r': command not found

Of course it is grails, because grails brings this gradlew file into existence via create-app

Speculating on how this is implemented... I'm guessing that at some time these files came out of git with the wrong git core.autocrlf and/or eol settings appropriate for a bash script which should never contain line feed characters.  Which it does:

```
od -c gradlew | head -1
0000000   #   !   /   b   i   n   /   s   h  \r  \n  \r  \n   #  \r  \n
```

> Can you try `./gradlew.bat bootRun`?

Yes that works, because CMD accepts \r. bash does not accept it, so grails should never create a file that won't work in bash.

**Comment by puneetbehl** [comment]
I apologize for any misunderstandings. To clarify my earlier statement, I am not sure if the following error is related to Grails. What I meant here is that the Grails Core does not have anything do with it. The Gradle Wrapper is generated using the `gradle wrapper --gradle-version 7.6.2` command. We do not customize it manually. If you think that the wrapper shipped with Grails Forge (or the newly created Grails App) is corrupted somewhere. Please feel free to submit a pull-request to update it [here](https://github.com/grails/grails-forge/tree/6.0.x/grails-forge-core/src/main/resources/gradle)

We're keen to understand the procedure for generating a fresh wrapper, as manual adjustments are not within our intentions. If you could kindly share the steps involved, it would be greatly appreciated.

**Comment by chrisbitmead** [comment]
@puneetbehl I know what you're saying. However, and sdkman installed gradle 7.6.2 and gradlew created gradlew has the right line endings. The gradlew at the location you point to above seems to have the right line endings. and yet grails create-app creates the wrong ones.

I can't actually see where in the code it calls gradlew wrapper --gradle-version. 

I would note though that CreateAppCommand.groovy contains this code:
```
        buildGradleFiles.each { File srcFile ->
            File destFile = new File(getDestinationDirectory(srcFile), BUILD_GRADLE)
            destFile.text = destFile.getText(ENCODING) + System.lineSeparator() + srcFile.getText(ENCODING)
        }
```

Which appears to be a bug that causes build.gradle to be mostly \n, and yet end in \r\n on Windows. A different thing I know. I don't know where it calls gradlew so I can't speculate on what the problem might be. Certainly I can run gradle wrapper --gradle-version 7.6.2 and it creates correct files.

**Comment by puneetbehl** [comment]
I am glad you found the problem. Thank you for looking into this. Would you be willing to send a pull-request to fix the `CreateAppCommand`?

**Comment by chrisbitmead** [comment]
What do you mean, I didn't find the problem. I did notice another minor problem that's all.

**Comment by puneetbehl** [comment]
Oh, I misunderstood. I will look into this and try to fix it with Grails 6.0.1. Can you please try if the problem persists when you create an application from https://start.grails.org selecting version 6.0.1-SNAPSHOT?

**Comment by xpusostomos** [comment]
The problem does not occur on start.grails.org, however I'm guessing that runs Linux and almost certainly this bug is only going to manifest on Windows. Why, I don't know.
